qa/sc_evm_forger_and_delegator_rewards.py

```diff
@@ -0,0 +1,525 @@
+        forger_5_address, forger_5_blockSignPubKey, forger_5_vrfPubKey = self.create_forger_stake(mc_node, sc_node_5, 13, 11)
+        forger_6_address, forger_6_blockSignPubKey, forger_6_vrfPubKey = self.create_forger_stake(mc_node, sc_node_6, 13, 11)
+        forger_7_address, forger_7_blockSignPubKey, forger_7_vrfPubKey = self.create_forger_stake(mc_node, sc_node_7, 13, 11)
+        self.sc_sync_all()
+
+        # we now have 7 stakes, one from creation and 6 just added
+        stakeList = sc_node_1.transaction_allForgingStakes()["result"]['stakes']
+        assert_equal(7, len(stakeList))
+
+        # reach fork point 1.3 and execute stake v1 upgrade
+        generate_next_block(sc_node_1, "first", force_switch_to_next_epoch=True)  # 69
+        generate_next_block(sc_node_1, "first", force_switch_to_next_epoch=True)  # 70
+        # Execute upgrade
+        old_forger_native_contract = SmartContract("ForgerStakes")
+        method = 'upgrade()'
+        contract_function_call(sc_node_1, old_forger_native_contract, FORGER_STAKE_SMART_CONTRACT_ADDRESS,
+                               evm_address_sc_node_1, method)
+
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=True)  # 71
+        self.sc_sync_all()
+
+        # Assert block creation by each node
+        generate_next_block(sc_node_2, "second")
+        generate_next_block(sc_node_3, "third")
+        generate_next_block(sc_node_4, "fourth")
+        generate_next_block(sc_node_5, "fifth")
+        generate_next_block(sc_node_6, "sixth")
+        generate_next_block(sc_node_7, "seventh")
+
+        # Reach epoch 79
+        self.advance_to_epoch(79)
+        # Send 100 zen to forger pool mc reward address
+        ft_pool_amount = 1000
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      mc_node,
+                                      format_eoa(FORGER_POOL_RECIPIENT_ADDRESS),
+                                      ft_pool_amount,
+                                      mc_return_address=mc_node.getnewaddress(),
+                                      generate_block=False)
+        mc_node.generate(self.withdrawalEpochLength - 7)
+        self.sync_all()
+        self.sc_sync_all()
+        sc_last_we_block_id = generate_next_block(sc_node_1, "third node")
+        self.sc_sync_all()
+
+        # Check that the fee distribution took place, exact fees before 1.4 are not tested here
+        api_fee_payments = http_block_getFeePayments(sc_node_1, sc_last_we_block_id)['feePayments']
+        # 7 because the genesis block is counted towards node 1 forger address, not reward address
+        assert_equal(7, len(api_fee_payments))
+
+        # trigger cert submission
+        # Generate 2 SC blocks on SC node and start them automatic cert creation.
+        generate_next_block(sc_node_1, "first node")  # 1 SC block to reach the end of WE
```

Comment:
isn't the comment misleading? we should have reached already the end of WE with the previous block, at line 204

Reply:
I don't really know how this works, copied this part from other test. Will remove the comment completely